src/main/java/org/pesmypetcare/webservice/dao/MedicationDaoImpl.java

```diff
@@ -0,0 +1,208 @@
+                                 MedicationEntity medication) {
+        DocumentReference medicationRef = getMedicationsRef(owner, petName).document(dateName);
+        medicationRef.set(medication);
+    }
+
+    @Override
+    public void deleteByDateAndName(String owner, String petName, String dateName) {
+        DocumentReference medRef = getMedicationsRef(owner, petName).document(dateName);
+        medRef.delete();
+    }
+
+    @Override
+    public void deleteAllMedications(String owner, String petName) throws DatabaseAccessException {
+        CollectionReference medicationsRef = getMedicationsRef(owner, petName);
+        try {
+            ApiFuture<QuerySnapshot> future = medicationsRef.get();
+            List<QueryDocumentSnapshot> medicationDocuments = future.get().getDocuments();
+            for (QueryDocumentSnapshot medicationDocument : medicationDocuments) {
+                medicationDocument.getReference().delete();
+            }
+        } catch (InterruptedException | ExecutionException e) {
+            throw new DatabaseAccessException(DELFAIL_KEY, e.getMessage());
+        }
+    }
+
+    @Override
+    public MedicationEntity getMedicationData(String owner, String petName, String dateName)
+            throws DatabaseAccessException {
+        CollectionReference medicationsRef = getMedicationsRef(owner, petName);
+        DocumentReference medicationsRefDoc = medicationsRef.document(dateName);
+        ApiFuture<DocumentSnapshot> future = medicationsRefDoc.get();
+        DocumentSnapshot medicationDoc;
+        try {
+            medicationDoc = future.get();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new DatabaseAccessException(DELFAIL_KEY, e.getMessage());
+        }
+        if (!medicationDoc.exists()) {
+            throw new DatabaseAccessException(INVALID_MEDICATION_EXC, MEDICATION_DOES_NOT_EXIST_EXC);
+        }
+        return medicationDoc.toObject(MedicationEntity.class);
+    }
+
+    @Override
+    public List<Map<List<String>, Object>> getAllMedicationData(String owner, String petName)
+            throws DatabaseAccessException {
+        CollectionReference medicationsRef = getMedicationsRef(owner, petName);
+        List<Map<List<String>, Object>> externalList = new ArrayList<>();
+        ApiFuture<QuerySnapshot> future = medicationsRef.get();
+        List<QueryDocumentSnapshot> medicationDocuments;
+        try {
+            medicationDocuments = future.get().getDocuments();
+        } catch (InterruptedException | ExecutionException e) {
+            throw new DatabaseAccessException(DELFAIL_KEY, e.getMessage());
+        }
+        for (QueryDocumentSnapshot medicationDocument : medicationDocuments) {
+            externalList.add(getEachInternalList(medicationDocument));
+        }
+        return externalList;
+    }
+
+    private Map<List<String>, Object> getEachInternalList
+            (QueryDocumentSnapshot medicationDocument) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['method def lparen' has incorrect indentation level 12, expected level should be 4.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5348142)